### PR TITLE
fix(useSendDetails): update errored state properly on input change

### LINKED ISSUE #1051: Send modal does not update "insufficient funds" button when inputing the correct amount of funds
### Overview

When sending an amount that is more than the available balance, the send modal does not update "insufficient funds" button to "Continue" after inputing the correct amount in the modal.

### Reference

Recreations steps:
1. Go to app.shapeshift.com
2. Input an incorrect, or more than the balance, amount of coin / token
3. Delete the incorrect amount and input an amount you can afford
4. Notice that the “insufficient funds” button does not update to “Continue"

Screen Shot 2022-02-22 at 3 43 38 AM

### Acceptance Criteria

- When the amount is edited to an acceptable amount, the "Insufficient Funds" button should change to "Continue"

### Screenshots/Mockups

_No response_

### Ownership

- [X] If my bounty needs engineering or needs product I have added the respective labels on the right
- [X] As the sponsor of this bounty I will review the changes in a preview environment (ops/product) or review the PR (engineering)

# Bounty Hunters
- [Join our discord](https://discord.gg/shapeshift)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info

### PR DESCRIPTION
**Pull request by stackedq**
## Description

fixed wrong button title on Send modal if the input is dirty.

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #1051

## Testing

Open Send modal, enter amount higher than your balance and change it to something you can afford, you must be able to go to the next step.

### CODE REVIEW

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
Again nice one @stackedq 👏🏽 ! 
Tried it back and forth with valid and invalid send amounts, with enough or not enough gas, and that fixes... but with some gotchas.

This only works when going from a field with some invalid amount, to empty input and then re-typing something.
If you're trying to input some invalid amount, then remove one char so that it becomes valid, the error is still present.
Removing one more char than re-typing it again makes the input valid.

**Comment by stackedq** [comment]
Hmm, let me check

**Comment by stackedq** [comment]
@gomesalexandre  `values.cryptoAmount` is not in sync with input value, that's the cause, gonna fix it rn.

**Review by gomesalexandre — APPROVED** [review]
Nice! Tested it again and it fixes the case I've mentioned in the previous review 🎉

File: src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx
Comment (cjthompson) [inline src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx:278]:
```suggestion
          // Don't show an error message when the input is empty
```

File: src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx
Comment (cjthompson) [inline src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx:324]:
```suggestion
          // Remove existing error messages because the send amount is valid
```

**Review by cjthompson — APPROVED** [review]
Approved with comments.

File: src/components/TokenRow/TokenRow.tsx
Comment (stackedq) [reply src/components/TokenRow/TokenRow.tsx:69]: Exactly, before this order change, `FormContext` was one change behind, because `onInputChange` was running before `onChange` (which is for internal form context). 
I also used the `if` statement instead of logical and operator, because I think it'll improve code readability.

**Comment by gomesalexandre** [comment]
@stackedq one last thing before the merge button gets clicked, could you change the PR title to be more reflective of what this actually solves? This is not so much about the wrong title but rather the send form/input state not updating correctly.

Something like `fix(useSendDetails): update errored state properly on input change`

A more detailed description of what was fixed would also help, since it can be included in the squashed commit description.

**Comment by gomesalexandre** [comment]
@0xean ^

**Pull request merged**
